### ISSUE TITLE
Detaching variance prediction when computing SNR to reweight MSE 

### ISSUE THREAD

**Issue by BailiangJ** [Voldemort108X/hetero_uncertainty#1]
Hi @Voldemort108X ,

Thanks a lot for sharing the code for the great work!

I have a question regarding the [SNR computation](https://github.com/Voldemort108X/hetero_uncertainty/blob/0ecdac258845ee64ab42e839546c60f8343ccc90/src/voxelmorph/train.py#L310).

In the paper, the gradient of the predicted variance is stopped when used to compute the SNR. 
![image](https://github.com/user-attachments/assets/89d8d63a-85c6-48ea-bcc0-e9a1089a9f49)

However, in the code implementation, the `logsigma_image` is neither detached when passing to the `compute_SNR_from_variance` nor detached in the function. 

Is this an error or the actual implementation?
Looking forward to your reply!
Thanks a lot!

Cheers,
Bailiang

**Comment by Voldemort108X** [comment]
Hi Bailing,

Thanks so much for your interest in this work! In our implementation, we detached it from the loss function after computing the SNR weighting:

https://github.com/Voldemort108X/hetero_uncertainty/blob/0ecdac258845ee64ab42e839546c60f8343ccc90/src/voxelmorph/voxelmorph/torch/losses.py#L20C1-L38C1

Please let me know if this addresses the question. 

Best,
Xiaoran

**Comment by BailiangJ** [comment]
Hi Xiaoran @Voldemort108X ,

Thanks a lot for the prompt reply!

Yes, I missed the part where the weights are detached in the WeightedMSE!

Thanks a lot!

Cheers,
Bailiang
